### ISSUE TITLE
[bitnami/postgresql] helm installation: issue with existingSecret: the related secret data is not correctly processed

### ISSUE THREAD

**Issue by ArtGee** [bitnami/charts#15731]
### Name and Version

bitnami/postgresql 12.2.6

### What architecture are you using?

amd64

### 1. Create a secret in default namespace with password:
```
kubectl create secret generic \
	servicedb-secret \
	--from-literal=postgres-password='XXX' \
	--from-literal=password='XXX'
```
### 2. Substitute the corresponding parameter values in values.yaml. The participated parameters:
```
global.postgresql.auth.existingSecret
global.postgresql.auth.secretKeys.adminPasswordKey
global.postgresql.auth.secretKeys.userPasswordKey

auth.secretKeys.adminPasswordKey
auth.secretKeys.userPasswordKey
```

### 3. Install a new Postgresql instance by using the following command:
```
helm install \
	  servicedb \
	  bitnami/postgresql \
	  --values /var/projects/postgresql/servicedb/manifests/helm/values.yaml \
	  --debug --timeout 10m0s

```

### Are you using any custom parameters or values?

Yes, the following parameter values are used in values.yaml (both global.postgresql and auth scopes are used, because otherwise  helm returns 'The secret "servicedb-secret" does not contain the key ""'):

```
global:
  postgresql:
    auth:
      postgresPassword: ""
      username: ""
      password: ""
      database: ""
      existingSecret: "servicedb-secret"
      secretKeys:
        adminPasswordKey: postgres-password
        userPasswordKey: password
        replicationPasswordKey: ""
...
auth:
  enablePostgresUser: true
  postgresPassword: ""
  username: ""
  password: ""
  database: ""
  replicationUsername: repl_user
  replicationPassword: ""
  existingSecret: "servicedb-secret"
  secretKeys:
    adminPasswordKey: postgres-password
    userPasswordKey: password
    replicationPasswordKey: replication-password
```

### What is the expected behavior?

The chart should successfully read the secret and substitute the password in the parameter related to the login postgres.

### What do you see instead?

```
2023-03-25 20:21:41.227 +04 [66] FATAL:  password authentication failed for user "postgres"
2023-03-25 20:21:41.227 +04 [66] DETAIL:  User "postgres" has no password assigned.
```

### Additional information

I suppose that the clue to this issue could be in the file /bitnami/charts/tree/main/bitnami/postgresql/templates/secrets.yaml between the lines:

```
{{- if .Values.auth.enablePostgresUser }}
{{- $postgresPassword = include "common.secrets.passwords.manage" (dict "secret" (include "postgresql.secretName" .) "key" $.Values.auth.secretKeys.adminPasswordKey "providedValues" (list "global.postgresql.auth.postgresPassword" "auth.postgresPassword") "context" $) | trimAll "\"" | b64dec }}
{{- end }}

```

Below you can see the full log of the postgresql instance pod.

```
postgresql 16:21:32.67
postgresql 16:21:32.68 Welcome to the Bitnami postgresql container
postgresql 16:21:32.68 Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql 16:21:32.69 Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql 16:21:32.69
postgresql 16:21:32.75 INFO  ==> ** Starting PostgreSQL setup **
postgresql 16:21:32.78 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 16:21:32.81 INFO  ==> Loading custom pre-init scripts...
postgresql 16:21:32.83 INFO  ==> Initializing PostgreSQL database...
postgresql 16:21:32.84 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
postgresql 16:21:32.85 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
postgresql 16:21:32.96 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql 16:21:32.98 INFO  ==> Loading custom scripts...

postgresql 16:21:33.00 INFO  ==> ** PostgreSQL setup finished! **
postgresql 16:21:33.04 INFO  ==> ** Starting PostgreSQL **
2023-03-25 20:21:33.091 +04 [1] LOG:  pgaudit extension initialized
2023-03-25 20:21:33.180 +04 [1] LOG:  starting PostgreSQL 15.2 on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
2023-03-25 20:21:33.182 +04 [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-03-25 20:21:33.182 +04 [1] LOG:  listening on IPv6 address "::", port 5432
2023-03-25 20:21:33.195 +04 [1] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2023-03-25 20:21:33.292 +04 [56] LOG:  database system was shut down at 2023-03-25 20:21:25 +04
2023-03-25 20:21:33.344 +04 [1] LOG:  database system is ready to accept connections
2023-03-25 20:21:41.227 +04 [66] FATAL:  password authentication failed for user "postgres"
2023-03-25 20:21:41.227 +04 [66] DETAIL:  User "postgres" has no password assigned.
        Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
2023-03-25 20:21:51.207 +04 [73] FATAL:  password authentication failed for user "postgres"
2023-03-25 20:21:51.207 +04 [73] DETAIL:  User "postgres" has no password assigned.
        Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
```

**Comment by ArtGee** [comment]
I have discovered that this is the same issue as [13386](https://github.com/bitnami/charts/issues/13386).
The problem is with the custom pg_hba.conf.
But there is still an issue with the global scope of the secret names when we set global.postgresql.auth.existingSecret but leave auth.secretKeys.adminPasswordKey empty. The output of the helm installation process is:
'The secret "servicedb-secret" does not contain the key ""' still remains.
